README.md
# [Optical Music Recognition Datasets](https://apacha.github.io/OMR-Datasets/)

This repository contains a collection of many datasets used for various Optical Music Recognition tasks, including
staff-line detection and removal, training of Convolutional Neuronal Networks (CNNs) or validating existing systems by
comparing your system with a known ground-truth.

Note that most datasets have been developed by researchers and using their dataset requires accepting a certain license
and/or citing their respective publications, as indicated for each dataset. Most datasets link to the official website,
where you can download the dataset.

If you are interested in Optical Music Recognition research, you can find a curated bibliography
at [https://omr-research.github.io/](https://omr-research.github.io/).

## Overview

The following datasets are referenced from this repository:

| Name                                          	                                                                           |    Engraving          	    |               Size              									               |                            Format             								                            |                            Typical usages              						                            |
|---------------------------------------------------------------------------------------------------------------------------|:--------------------------:|:-------------------------------------------------------:|:---------------------------------------------------------------------------------:|:----------------------------------------------------------------------------------------:|
| [Handwritten Online Musical Symbols (HOMUS)](#handwritten-online-musical-symbols-homus) 	                                 |  Handwritten           	   |             15200 symbols   	    									              |                          Text-File             								                           |                     Symbol Classification (online + offline)  						                     |
| [Universal Music Symbol Collection](#universal-music-symbol-collection)                 	                                 |  Typeset + Handwritten 	   |              ~ 90000 symbols	    									              |                          Images                								                           |                     Symbol Classification (offline)           						                     |
| [CVC-MUSCIMA](#cvc-muscima)                                                                 	                             |  Handwritten           	   |             1000 score images     									             |                          Images                								                           |                     Staff line removal, writer identification 						                     |
| [MUSCIMA++](#muscima)                                                                       	                             |  Handwritten           	   |             > 90000 annotatations 									             |                 Images, Measure Annotations, MuNG            			                  |  Symbol Classification, Object Detection, End-To-End Recognition, Measure Recognition 	  |
| [Mashcima](#mashcima)                                                                       	                             |  Handwritten           	   |                   unlimited 									                   |                               Images            			                               |                                        Various 	                                         |
| [DeepScores V1](#deepscores-v1)                                                                    	                      |  Typeset               	   |              300000 images 	    									               |                         Images, XML              								                         |            Symbol Classification, Object Detection, Semantic Segmentation  	             |
| [DeepScores V2](#deepscores-v2)                                                                    	                      |  Typeset               	   |              255385 images 	    									               |                         Images, XML              								                         |            Object Detection, Semantic Segmentation, Instance Segmentation  	             |
| [DoReMi](#doremi)                                                                    	                                    |  Typeset               	   |               6432 images 	    									                |              Images, XML, musicXML, MEI, MIDI              								               | Symbol Classification, Object Detection, Semantic Segmentation, Instance Segmentation  	 |
| [PrIMuS](#primus)                                                                         	                               |  Typeset               	   |              87678 incipits 	    									              |              Images, MEI, Simplified encoding, agnostic encoding  	               |               End-to-End Recognition                                   			               |
| [Baro Single Stave Dataset](#baró-single-stave-dataset)                                     	                             | Handwritten              	 |                95 images 	    									                 |                          Images, Simplified encoding  	                           |                        End-to-End Recognition                 			                        |
| [Multimodal Sheet Music Dataset](#multimodal-sheet-music-dataset)                           	                             |  Typeset               	   |              497 songs      	    									              |                   Images, MIDI, Lilypond, MuNG (noteheads) 				                   |              End-to-End Recognition, Multimodal Retrieval, Score Following               |
| [Sheet Midi Retrieval Dataset](#sheet-midi-retrieval-dataset)                             	                               |  Typeset               	   |              200 songs      	    									              |                   Images (Jpg and PDF), MIDI, CSV         				                    |                          Multimodal Retrieval, Score Following                           |
| [AudioLabs v1](#audiolabs-v1)                                                                                             |   Typeset              	   |      940 score images; 24,329 bounding boxes	    	      |                         Images                   							                          |                   Box Annotation Detection                     						                    |
| [AudioLabs v2](#audiolabs-v2)                                                                                             |          Typeset           |      940 score images; 85,980 bounding boxes	    	      |                         Images                   							                          |                   Box Annotation Detection                     						                    |
| [AudioLabs v2](#audiolabs-v2)                                                                                             |          Typeset           |      940 score images; 85,980 bounding boxes	    	      |                         Images                   							                          |                   Box Annotation Detection                     						                    |
| [OpenScore Lieder](#openscore-lieder)                                                                 	                   |  Typeset               	   |                1356 files 	    									                |                                MuseScore 								                                 |                     Various                                   						                     |
| [OpenScore String Quartets](#openscore-string-quartets)                                                                 	 |  Typeset               	   |                106 files 	    									                 |                                 MuseScore								                                 |                     Various                                   						                     |
| [MuseScore](#musescore)                                                                 	                                 |  Typeset               	   |              > 340000 files 	    									              |                         MuseScore, PDF, MusicXML 								                         |                     Various                                   						                     |
| [MuseScore Monophonic MusicXML Dataset](#musescore-monophonic-musicxml-dataset)         	                                 |  Typeset               	   |              17000 IDs      	    									              |                         IDs for MuseScore files  								                         |                     Various                                   						                     |
| [Capitan collection](#capitan-collection)                                               	                                 | Handwritten              	 |             10230 symbols         									             |                         Images, Text-File    	    							                         |                     Symbol Classification                     						                     |
| [SEILS Dataset](#seils-dataset)                                               	                                           |   Typeset              	   | 30 madrigals, 150 original images, 930 symbolic files 	 | Images (PDF), .ly, .mid, .xml, .musx, .krn, .mei, .mns, .agnostic, .semantic    	 |                                  Various         						                                  |
| [Rebelo Dataset](#rebelo-dataset)                                                        	                                |  Typeset               	   |              15000 symbols  	    									              |                          Images                								                           |                     Symbol Classification                     						                     |
| [Fornes Dataset](#fornes-dataset)                                                       	                                 |  Handwritten           	   |              4100 symbols   	    									              |                          Images                 								                          |                     Symbol Classification                     						                     |
| [Choi Accidentals Dataset](#choi-accidentals-dataset)                                                         	           |  Typeset               	   |                2955 images	    									                |                Images with special filename          	    							                 |                              Symbol Classification   						                              |
| [Audiveris OMR](#audiveris-omr)                                                         	                                 |  Typeset               	   |              800 annotations	    									              |                         Images, XML          	    							                         |                     Symbol Classification, Object Detection   						                     |
| [Printed Music Symbols Dataset](#printed-music-symbols-dataset)                         	                                 |  Typeset               	   |              200 symbols    	    									              |                          Images                								                           |                     Symbol Classification                     						                     |
| [Music Score Classification Dataset](#music-score-classification-dataset)               	                                 |  Typeset               	   |             1000 score images     									             |                          Images                								                           |                     Sheet Classification                      						                     |
| [OpenOMR Dataset](#openomr-dataset)                                                     	                                 |  Typeset               	   |              706 symbols    	    									              |                          Images                								                           |                     Symbol Classification                     						                     |
| [Gamera MusicStaves Toolkit](#gamera-musicstaves-toolkit)                                	                                |  Typeset               	   |              32 score images	    									              |                          Images                								                           |                     Staff line removal                        						                     |
| [Early Typographic Prints](#early-typographic-prints)                                   	                                 |  Typeset               	   |             240 score images      									             |                                     								                                      |                                          						                                          |
| [Silva Online Handwritten Symbols](#silva-online-handwritten-symbols)                   	                                 |  Handwritten           	   |              12600 symbols  	    									              |                                     								                                      |                                          						                                          |
| [IMSLP](#imslp)                                                                         	                                 |  Typeset               	   |             >420000 score images  									             |                         PDF                      								                         |                   Various                                      						                    |
| [Byrd Dataset](#byrd-dataset)                                                           	                                 |  Typeset               	   |              34 score images	    									              |                         Images                   								                         |                   Various                                      						                    |
| [MScoreLib Dataset](#MScoreLib)                                                           	                               |  Typeset               	   |          ~6000 MusicXML scores 	    									           |                        MusicXML                   								                        |                   Various                                      						                    |

If you find mistakes or know of any relevant datasets, that are missing in this list,
please [open an issue](https://github.com/apacha/OMR-Datasets/issues/new) or directly file a pull request.

## Tools for working with the datasets

A collection of tools that simplify the downloading and handling of datasets used for Optical Music Recognition (OMR).
These tools are available as Python package ``omrdatasettools`` on PyPi.

[![Build Status](https://github.com/apacha/OMR-Datasets/actions/workflows/python-app.yml/badge.svg)](https://travis-ci.org/apacha/OMR-Datasets) [![codecov](https://codecov.io/gh/apacha/OMR-Datasets/branch/master/graph/badge.svg)](https://codecov.io/gh/apacha/OMR-Datasets) [![PyPI version](https://badge.fury.io/py/omrdatasettools.svg)](https://badge.fury.io/py/omrdatasettools) [![Documentation Status](https://readthedocs.org/projects/omr-datasets/badge/?version=latest)](http://omr-datasets.readthedocs.io/en/latest/?badge=latest) [![GitHub license](https://img.shields.io/badge/License-MIT-brightgreen.svg)](https://raw.githubusercontent.com/apacha/OMR-Datasets/master/LICENSE.txt)

# Handwritten Online Musical Symbols (HOMUS)

**Official website**: [http://grfia.dlsi.ua.es/homus/](http://grfia.dlsi.ua.es/homus/)

[![License](https://img.shields.io/badge/License-Unknown-red.svg)](http://grfia.dlsi.ua.es/homus/)

**Summary**: The Handwritten Online Musical Symbols (HOMUS) dataset is a reference corpus with around 15000 samples for
research on the recognition of online handwritten music notation. For each sample, the individual strokes that the
musicians wrote on a Samsung Tablet using a stylus were recorded and can be used in online and offline scenarios.

**Scientific Publication**: J. Calvo-Zaragoza and J. Oncina, "Recognition of Pen-Based Music Notation: The HOMUS
Dataset," 2014 22nd International Conference on Pattern Recognition, Stockholm, 2014, pp.
3038-3043. [DOI: 10.1109/ICPR.2014.524](http://dx.doi.org/10.1109/ICPR.2014.524)

**Example**:

![Example of HOMUS dataset](samples/homus.png)

*Remarks*: The original dataset contains around 20 artifacts and misclassifications that were reported to the authors
and [corrected by Alexander Pacha](https://github.com/apacha/Homus).

# Universal Music Symbol Collection

**Official website**: [https://github.com/apacha/MusicSymbolClassifier](https://github.com/apacha/MusicSymbolClassifier), [Slides](https://docs.google.com/presentation/d/14g97TnrcI9o-5D6DIY-dMFfBp9kUAqpbe86c-VE83Bk/edit?usp=sharing)

[![License: MIT](https://img.shields.io/badge/License-MIT-brightgreen.svg)](https://opensource.org/licenses/MIT)

**Summary**: A collection of various other datasets which combines 7 datasets into a large unified dataset of 90000 tiny
music symbol images from 79 classes that can be used to train a universal music symbol classifier. 74000 symbols are
handwritten and 16000 are printed symbols.

**Scientific Publication**: Alexander Pacha, Horst Eidenberger. Towards a Universal Music Symbol Classifier. Proceedings
of the 12th IAPR International Workshop on Graphics Recognition, Kyoto, Japan, November 2017. [DOI: 10.1109/ICDAR.2017.265](http://dx.doi.org/10.1109/ICDAR.2017.265)

**Example**:

![Example of MusicSymbolClassifier dataset](samples/universal-music-symbol-collection.png)

# CVC-MUSCIMA

**Official website**: [http://www.cvc.uab.es/cvcmuscima/index_database.html](http://www.cvc.uab.es/cvcmuscima/index_database.html)

[![License: CC BY-NC-SA 4.0](https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-blue.svg)](https://creativecommons.org/licenses/by-nc-sa/4.0/)

**Summary**: The CVC-MUSCIMA database contains handwritten music score images, which has been specially designed for
writer identification and staff removal tasks. The database contains 1,000 music sheets written by 50 different
musicians. All of them are adult musicians, in order to ensure that they have their own characteristic handwriting
style. Each writer has transcribed the same 20 music pages, using the same pen and the same kind of music paper (with
printed staff lines). The set of the 20 selected music sheets contains music scores for solo instruments and music
scores for choir and orchestra.

**Scientific Publication**: Alicia Fornés, Anjan Dutta, Albert Gordo, Josep Lladós. CVC-MUSCIMA: A Ground-truth of
Handwritten Music Score Images for Writer Identification and Staff Removal. International Journal on Document Analysis
and Recognition, Volume 15, Issue 3, pp 243-251, 2012. [DOI: 10.1007/s10032-011-0168-2](http://dx.doi.org/10.1007/s10032-011-0168-2)

**Example**:

![Example of CVC MUSCIMA dataset](samples/cvc-muscima.png)

# MUSCIMA++

**Official website**: [https://ufal.mff.cuni.cz/muscima](https://ufal.mff.cuni.cz/muscima)

**Current development**: [https://github.com/OMR-Research/muscima-pp](https://github.com/OMR-Research/muscima-pp)

[![License: CC BY-NC-SA 4.0](https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-blue.svg)](https://creativecommons.org/licenses/by-nc-sa/4.0/)

**Summary**: MUSCIMA++ is a dataset of handwritten music notation for musical symbol detection that is based on the
MUSCIMA dataset. It contains 91255 symbols, consisting of both notation primitives and higher-level notation objects,

CHANGES.md
Changelog
=========
1.4.0
-----
Adding several new datasets, upgrading to Python 3.11, and refactoring downloader to use Pathlib and support tar-gz.

1.3.1
-----
Bugfixing an error that generates incorrect MUSCIMA++ image masks because 
images were incorrectly associated with each other.

1.3.0
-----
Added download capabilities of DeepScores V1 with extended vocabulary and
opened the downloader, so you can download custom datasets, as well as utilize
other methods from it that were previously private.

1.2.2
-----
Fixed incorrect import statement in `__init__.py`

1.2.1
-----
Fixed dependency problem during `setup.py` that prevented the package from being 
installed if the dependent libraries are not yet installed (which defeats
the purpose of declaring dependencies in setup.py).
Changing to semantic versioning with three numbers. 
1.2 is now considered 1.2.0.

1.2
---
Attempting to declare dependencies in `setup.py` properly

1.1
---
Updated MuscimaPlusPlusSymbolImageGenerator to work with MUSCIMA++ 2.0.
Added quality-of-life improvement suggested by @yvan674 to make importing 
common classes such as the downloader easier.


1.0
---
Dramatically simplified the tools for downloading datasets. 
Removed mostly unused code and re-organized project structure and documentation.

0.19
----
New Image generator that can take MUSCIMA++ v2.0 images and 
generate masks for instance segmentation of staffs, as well as
masks for semantic segmentation for all objects.

0.18
----
Changing MUSCIMA++ Downloader to accept a string instead of integer for enabling
future versioning of the dataset beyond integers, e.g., "2.1".

Previous releases
-----------------
For information on previous releases, check out the [Github Repository](https://github.com/apacha/OMR-Datasets/releases)

omrdatasettools/OmrDataset.py
from enum import Enum, auto
from typing import Dict


class OmrDataset(Enum):
    """
    The available OMR datasets that can be automatically downloaded with Downloader.py
    """

    Audiveris = auto()
    """ 
    The Audiveris OMR dataset from https://github.com/Audiveris/omr-dataset-tools,
    Copyright 2017 by Hervé Bitteur under AGPL-3.0 license """

    Baro = auto()
    """
    The Baro Single Stave dataset from http://www.cvc.uab.es/people/abaro/datasets.html, Copyright 2019 Arnau Baró,
    Pau Riba, Jorge Calvo-Zaragoza, and Alicia Fornés under CC-BY-NC-SA 4.0 license
    """

    Capitan = auto()
    """
    The Capitan dataset from http://grfia.dlsi.ua.es/, License unspecified, free for research purposes
    """

    CvcMuscima_MultiConditionAligned = auto()
    """
    Custom version of the CVC-MUSCIMA dataset that contains all images in grayscale, binary and with the
    following staff-line augmentations: interrupted, kanungo, thickness-variation-v1/2, y-variation-v1/2
    typeset-emulation and whitespeckles. (all data augmentations that could be aligned automatically).
    The grayscale images are different from the WriterIdentification dataset, in such a way, that they were aligned
    to the images from the Staff-Removal dataset. This is the recommended dataset for object detection, as the
    MUSCIMA++ annotations can be used with a variety of underlying images.
    See https://github.com/apacha/CVC-MUSCIMA to learn more.
    """

    CvcMuscima_StaffRemoval = auto()
    """
    The larger version of the CVC-MUSCIMA dataset for staff removal in black and white with augmentations
    from http://www.cvc.uab.es/cvcmuscima/index_database.html,
    Copyright 2012 Alicia Fornés, Anjan Dutta, Albert Gordo and Josep Lladós under CC-BY-NC-SA 4.0 license
    """

    CvcMuscima_WriterIdentification = auto()
    """
    The smaller version of the CVC-MUSCIMA dataset for writer identification in grayscale
    from http://www.cvc.uab.es/cvcmuscima/index_database.html,
    Copyright 2012 Alicia Fornés, Anjan Dutta, Albert Gordo and Josep Lladós under CC-BY-NC-SA 4.0 license
    """

    Edirom_Bargheer = auto()
    """
    Edirom dataset. All rights reserved
    """

    Edirom_FreischuetzDigital = auto()
    """
    Edirom datasets on Freischuetz from https://freischuetz-digital.de/edition.html. All rights reserved.
    """

    Fornes = auto()
    """
    The Fornes Music Symbols dataset from http://www.cvc.uab.es/~afornes/, License unspecified - citation requested
    """

    Homus_V1 = auto()
    """
    The official HOMUS dataset from http://grfia.dlsi.ua.es/homus/, License unspecified.
    """

    Homus_V2 = auto()
    """
    The improved version of the HOMUS dataset with several bugs-fixed from https://github.com/apacha/Homus.
    """

    MuscimaPlusPlus_V1 = auto()
    """
    The MUSCIMA++ dataset from https://ufal.mff.cuni.cz/muscima, Copyright 2017 Jan Hajic jr.
    under CC-BY-NC-SA 4.0 license.
    """

    MuscimaPlusPlus_V2 = auto()
    """
    The second version of the MUSCIMA++ dataset from https://github.com/OMR-Research/muscima-pp.
    """

    MuscimaPlusPlus_Images = auto()
    """
    The subset of 140 images from the CVC-MUSCIMA dataset that were used for the MUSCIMA++ dataset.
    """

    MuscimaPlusPlus_MeasureAnnotations = auto()
    """
    A sub-set of the MUSCIMA++ annotations that contains bounding-box annotations for staves, staff measures and
    system measures. It was semi-automatically constructed from existing annotations and manually verified for
    correctness. The annotations are available in a plain JSON format as well as in the COCO format.
    """

    OpenOmr = auto()
    """
    The OpenOMR Symbols dataset from https://sourceforge.net/projects/openomr/, Copyright 2013 by Arnaud F.
    Desaedeleer under GPL license.
    """

    Printed = auto()
    """
    The Printed Music Symbols dataset from https://github.com/apacha/PrintedMusicSymbolsDataset, Copyright 2017 by
    Alexander Pacha under MIT license.
    """

    Rebelo1 = auto()
    """
    The Rebelo dataset (part 1) with music symbols from http://www.inescporto.pt/~arebelo/index.php, Copyright 2017
    by Ana Rebelo under CC BY-SA 4.0 license
    """

    Rebelo2 = auto()
    """
    The Rebelo dataset (part 2) with music symbols from http://www.inescporto.pt/~arebelo/index.php, Copyright 2017
    by Ana Rebelo under CC BY-SA 4.0 license
    """

    DeepScores_V1_Extended_100_Pages = auto()
    """
    Subselection of 100 pages from the DeepScore dataset (version 1) with extended vocabulary from 
    https://tuggeluk.github.io/downloads/, License unspecified.
    """

    DeepScores_V1_Extended = auto()
    """
    The DeepScore dataset (version 1) with extended vocabulary from https://tuggeluk.github.io/downloads/,
    License unspecified.
    """

    DeepScores_V2_Dense = auto()
    """
    Subselection of 1714 pages from the DeepScore dataset (version 2) with extended vocabulary from 
    https://zenodo.org/records/4012193, under CC BY 4.0 license.
    """

    DeepScores_V2_Complete = auto()
    """
    The complete DeepScore dataset (version 2) from 
    https://zenodo.org/records/4012193, under CC BY 4.0 license.
    
    WARNING: The size of this dataset is over 80GB! 
    """

    AudioLabs_v1 = auto()
    """
    The AudioLabs v1 dataset (aka. Measure Bounding Box Annotation) from
    https://www.audiolabs-erlangen.de/resources/MIR/2019-ISMIR-LBD-Measures, Copyright 2019 by Frank Zalkow, Angel
    Villar Corrales, TJ Tsai, Vlora Arifi-Müller, and Meinard Müller under CC BY-NC-SA 4.0 license.
    """

    AudioLabs_v2 = auto()
    """
    The AudioLabs v2 dataset, enhanced with staves, staff measures and the original system measures. The annotations
    are available in csv, JSON and COCO format.
    """

    ChoiAccidentals = auto()
    """
    The Accidentals detection dataset by Kwon-Young Choi from https://www-intuidoc.irisa.fr/en/choi_accidentals/,
    License unspecified.
    """

    DoReMi = auto()
    """
    DoReMi dataset from https://github.com/steinbergmedia/DoReMi/, License unspecified.
    """

    OpenScoreLieder = auto()
    """
    OpenScore Lieder corpus from https://github.com/OpenScore/Lieder, CC-0 license.
    """

    OpenScoreStringQuartets = auto()
    """
    OpenScore StringQuartet corpus from https://github.com/OpenScore/StringQuartets, CC-0 license.
    """

    MScoreLib_All = auto()
    """
    The full MScoreLib corpus from http://mscorelib.com/, manually inputed scores by humans, License unspecified.
    """

    MScoreLib_Scriabin = auto()
    """
    MScoreLib corpus of Scriabin music, converted with SharpEye and PhotoScore, from http://mscorelib.com/, 
    License unspecified.
    """

    MScoreLib_Prokofiev = auto()
    """
    MScoreLib corpus of Prokofiev music, converted with SharpEye and PhotoScore, from http://mscorelib.com/, 
    License unspecified.
    """

    def get_dataset_download_url(self) -> str:
        """ Returns the url of the selected dataset.
            Example usage: OmrDataset.Fornes.get_dataset_download_url() """
        return self.dataset_download_urls()[self.name]

    def get_dataset_filename(self) -> str:
        """ Returns the name of the downloaded zip file of a dataset.
            Example usage: OmrDataset.Fornes.get_dataset_filename() """
        dataset_url = self.get_dataset_download_url()
        dataset_filename = dataset_url.split("/")[-1]
        return dataset_filename

    def dataset_download_urls(self) -> Dict[str, str]:
        """ Returns a mapping with all URLs, mapped from their enum keys """
        return {
            # Official URL: https://github.com/Audiveris/omr-dataset-tools/tree/master/data/input-images
            "Audiveris": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/AudiverisOmrDataset.zip",

            # Official URL: http://www.cvc.uab.es/people/abaro/datasets/MUSCIMA_ABARO.zip
            "Baro": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/BaroMuscima.zip",

            # Official URL: http://grfia.dlsi.ua.es/cm/projects/timul/databases/BimodalHandwrittenSymbols.zip
            "Capitan": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                       "BimodalHandwrittenSymbols.zip",

            # Official URL: http://www.cvc.uab.es/cvcmuscima/CVCMUSCIMA_WI.zip
            "CvcMuscima_WriterIdentification": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                               "CVCMUSCIMA_WI.zip",

            # Official URL: http://www.cvc.uab.es/cvcmuscima/CVCMUSCIMA_SR.zip
            "CvcMuscima_StaffRemoval": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                       "CVCMUSCIMA_SR.zip",

            # Official URL: https://github.com/apacha/CVC-MUSCIMA
            "CvcMuscima_MultiConditionAligned": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                                "CVCMUSCIMA_MCA.zip",

            "Edirom_Bargheer": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/Bargheer.zip",

            "Edirom_FreischuetzDigital": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                         "FreischuetzDigital.zip",

            # Official URL: http://www.cvc.uab.es/cvcmuscima/datasets/Music_Symbols.zip
            "Fornes": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/Music_Symbols.zip",

            # Official URL: http://grfia.dlsi.ua.es/homus/HOMUS.zip
            "Homus_V1": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/HOMUS.zip",

            # Official URL: https://github.com/apacha/Homus
            "Homus_V2": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/HOMUS-2.0.zip",

            # Official URL:
            # https://lindat.mff.cuni.cz/repository/xmlui/bitstream/handle/11372/LRT-2372/MUSCIMA-pp_v1.0.zip?sequence=1&isAllowed=y
            "MuscimaPlusPlus_V1": "https://github.com/OMR-Research/muscima-pp/releases/download/v1.0/"
                                  "MUSCIMA-pp_v1.0.zip",

            # Official URL: https://github.com/OMR-Research/muscima-pp
            "MuscimaPlusPlus_V2": "https://github.com/OMR-Research/muscima-pp/releases/download/v2.0/MUSCIMA-pp_v2.0.zip",

            "MuscimaPlusPlus_Images": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                      "CVC_MUSCIMA_PP_Annotated-Images.zip",

            "MuscimaPlusPlus_MeasureAnnotations": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                                  "MUSCIMA-pp_v1.0-measure-annotations.zip",

            # Official URL: https://sourceforge.net/projects/openomr/
            "OpenOmr": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/OpenOMR-Dataset.zip",

            "Printed": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/PrintedMusicSymbolsDataset.zip",

            "Rebelo1": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                       "Rebelo-Music-Symbol-Dataset1.zip",
            "Rebelo2": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                       "Rebelo-Music-Symbol-Dataset2.zip",

            # Official URL: "https://repository.cloudlab.zhaw.ch/artifactory/deepscores/ds_extended.zip",
            "DeepScores_V1_Extended_100_Pages": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                                "deep-scores-v1-extended-100pages.zip",
            "DeepScores_V1_Extended": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                      "deep-scores-v1-extended.zip",

            # Official URL: https://zenodo.org/records/4012193
            "DeepScores_V2_Dense": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                   "deep-scores-v2-dense.tar.gz",
            "DeepScores_V2_Complete": "https://zenodo.org/records/4012193/files/ds2_complete.tar.gz?download=1",

            # Official URL: https://www.audiolabs-erlangen.de/resources/MIR/2019-ISMIR-LBD-Measures
            "AudioLabs_v1": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/AudioLabs_v1.zip",

            "AudioLabs_v2": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/AudioLabs_v2.zip",

            # Official URL: https://www-intuidoc.irisa.fr/en/choi_accidentals/
            "ChoiAccidentals": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                               "choi_accidentals_dataset.zip",

            # Official URL: https://github.com/steinbergmedia/DoReMi/
            "DoReMi": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/DoReMi_v1.zip",

            # Official URL: https://github.com/OpenScore/Lieder
            "OpenScoreLieder": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                               "OpenScore-Lieder-Snapshot-2023-10-30.zip",

            # Official URL: https://github.com/OpenScore/StringQuartets
            "OpenScoreStringQuartets": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                       "OpenScore-StringQuartets-Snapshot-2023-10-30.zip",

            # Official URL: http://mscorelib.com/
            "MScoreLib_All": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/MScoreLib_all.zip",
            "MScoreLib_Scriabin": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                  "MScoreLib_Aleksandr_Scriabin.zip",
            "MScoreLib_Prokofiev": "https://github.com/apacha/OMR-Datasets/releases/download/datasets/"
                                   "MScoreLib_Sergey_Prokofiev.zip",


        }
